packages/oats-runtime/src/make.ts

```diff
@@ -284,10 +295,19 @@ export function makeBoolean(): Maker<unknown, boolean> {
   };
 }
 
-export function makeArray(maker: any, minSize?: number, maxSize?: number) {
-  return (value: any, opts?: MakeOptions) => {
-    if (!Array.isArray(value)) {
-      return getErrorWithValueMsg('expected an array', value);
+export function makeArray(
+  maker: any,
+  minSize?: number,
+  maxSize?: number
+): Maker<unknown, unknown[]> {
+  return (x, opts) => {
+    let value: unknown[];
+    if (Array.isArray(x)) {
+      value = x;
+    } else if (opts?.allowConvertForArrayType) {
```

Comment:
why we would want to do this?

Reply:
to support array schemas for query parameters
`param=a&param=b` is converted by koa to `query: {param: ['a', 'b']}`
but
`param=a` is converted to `query: {param: 'a'}`
So if we have an array schema for a query parameter, it'll fail unless we tell it to convert a single element to an array

Reply:
interesting... Idt we're supporting any other style but `form` as of now. So the fact that we're adding support for repeated query params and arrays (still for the same `form` style) isn't breaking anything in our code or against open api spec.
The only thing that i may be missing is this non-default `explode: false` option. Useless thing imho 😅 but I'll look into this (we do have it the code).

Reply:
> is this a must have for you?

no 😄 i didn't bring it up. Only agreed it's a good thing to address since we don't have a standard implementation for this in our services. Splitting by `,` goes only so far.

Reply:
i just checked, we're not supporting `explode=false` for queries. So we're only supporting the standard serialization.
There's this line
https://github.com/smartlyio/oats/blob/master/packages/oats/src/generate-types.ts#L244
but i'm not sure what it does because `explode` default value is true, so what is this check for? 🤔

Reply:
I would imagine that it deals with exploding the query params to an object. So you would have a single query param of type object and the incoming query object from the adapter is matched to that.

Reply:
> So you would have a single query param of type object
on server side query param is either a string or an array of strings, so how can it support a query param of type object?